### PR TITLE
Documentation - Add dependencies page

### LINKED ISSUE #8080: How to find which Addon/-s require a certain one ?
Each Mission launch, i get "No entry <<..>> .ace_ai'." message, how to find in the dozens of .pbo-s, which one/-s is requiring it ?

I didn't find the answer on the Wiki , so if its possible it should be added there . I'll highly appreciate help in solving the issue .

### PR DESCRIPTION
**Pull request by TyroneMF**
**When merged this pull request will:**
- Lists all components and it's required addons.
- Also gives a brief explanation of how to remove individual components without errors.
- resolves #8080

### CODE REVIEW

**Comment by bux** [comment]
I don't like that this is not automated.
You should either write your own (e.g. python) script to gather the information or make use / copy functionality from: https://github.com/acemod/ACE3/blob/master/docs/_includes/dependencies_list.md

**Comment by bux** [comment]
I don't like that this is not automated.
You should either write your own (e.g. python) script to gather the information or make use / copy functionality from: https://github.com/acemod/ACE3/blob/master/docs/_includes/dependencies_list.md

**Comment by jonpas** [comment]
Agreed with @bux. The existing script output might be able to be reused for this even.

**Comment by jonpas** [comment]
image

**Comment by jonpas** [comment]
All automated and cleaned up now.

File: docs/wiki/development/dependencies.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+Because `ace_zeus` is being removed you must also remove any components that require `ace_zeus` which in this case would be `ace_intelitems`.
+
+## 2. Dependencies
+
+{% assign pages_by_title = site.pages | sort: "title" %}
+{% for page in pages_by_title %}
+    {%- if page.group == 'feature' and page.component -%}
+        ### {{ page.title }}
+        {% capture component %}{{ page.component }}{% endcapture %}
+        {% include dependencies_list.md component=component %}
```
Comment (jonpas) [inline docs/wiki/development/dependencies.md:29]:
```suggestion
        {% include dependencies_list.md component=component %}
        {% if page.component == "common" or page.component == "main" or page.component == "interaction" or page.component == "interac_menu" %}
            _Note: This module is required by nearly all other modules. Do NOT remove it!_
        {% endif %}
```

**Review by jonpas — APPROVED** [review]

File: docs/wiki/development/dependencies.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+{% assign pages_by_title = site.pages | sort: "title" %}
+{% for page in pages_by_title %}
+    {%- if page.group == 'feature' and page.component -%}
+        ### {{ page.title }}
+
+        {% capture component %}{{ page.component }}{% endcapture %}
+        {% include dependencies_list.md component=component %}
+
+        {%- if page.component == "common" or page.component == "main" or page.component == "interaction" or page.component == "interact_menu" -%}
```
Comment (bux) [inline docs/wiki/development/dependencies.md:32]: maybe add a frontmatter property to these pages, and check for it here

Comment (jonpas) [reply docs/wiki/development/dependencies.md:32]: It's very specific to those 4 and that won't change without a big rework, so I don't really see the point. It's easier to miss a frontmatter variable than this too.

Comment (bux) [reply docs/wiki/development/dependencies.md:32]: Another might be added.
I don't like hardcoding these things here.
That's all. So please change it 👍

Comment (jonpas) [reply docs/wiki/development/dependencies.md:32]: So is it a "maybe add" or "do add" now? Your wording confuses me between idea inquiry and statement.

Comment (bux) [reply docs/wiki/development/dependencies.md:32]: Yes, please don't check for hardcoded strings in code. Please use a property for it.

**Review by bux — APPROVED** [review]
thank  you

**Comment by rautamiekka** [comment]
What does the removal of deps from the page solve ? That way one has to find the deps page and look through it. Even a link to the deps page woulda been better than nothing at all; nothing at all also makes it seem there's no modularity and/or deps.

**Comment by jonpas** [comment]
@rautamiekka https://github.com/acemod/ACE3/pull/8706#issuecomment-980362214

**Pull request merged**